tests/module.rs

```diff
@@ -0,0 +1,102 @@
+pub mod inner {
+    use snafu::Snafu;
+
+    #[derive(Debug)]
+    pub struct Dummy0;
+
+    #[derive(Debug)]
+    pub struct Dummy1;
+
+    #[derive(Debug)]
+    pub struct Dummy2;
+
+    #[derive(Debug)]
+    pub struct Dummy3;
+
+    #[derive(Debug, Snafu)]
+    #[snafu(module, visibility(pub))]
+    pub enum PubError {
+        Variant { v: Dummy0 },
+    }
+
+    #[derive(Debug, Snafu)]
+    #[snafu(module(custom_pub), visibility(pub))]
+    pub enum PubWithCustomModError {
+        Variant { v: Dummy1 },
+    }
+
+    #[derive(Debug, Snafu)]
+    #[snafu(module(custom_pub_crate), visibility(pub(crate)))]
+    pub(crate) enum PubCrateWithCustomModError {
+        Variant { v: Dummy2 },
+    }
+
+    mod child {
+        use super::Dummy3;
+        use snafu::Snafu;
+
+        #[derive(Debug, Snafu)]
+        #[snafu(module, visibility(pub(in crate::inner)))]
+        pub enum RestrictedError {
+            Variant { v: Dummy3 },
+        }
+    }
+
+    #[test]
+    fn can_set_module_visibility_restricted() {
+        let _ = self::child::restricted_error::VariantSnafu { v: Dummy3 }.build();
+    }
+}
+
+use self::inner::Dummy1;
+use snafu::Snafu;
+
+#[derive(Debug, Snafu)]
+#[snafu(module)]
+pub enum SomeError {
+    Variant { v: i32 },
+}
+
+#[derive(Debug, Snafu)]
+#[snafu(module)]
+pub enum QualifiedError {
+    Variant {
+        unqualified: Dummy1,
+        mod_struct: inner::Dummy0,
+        self_struct: self::Dummy1,
+        crate_struct: crate::Dummy1,
```

Comment:
Ah, I see you are importing them different ways here. Still, is there a benefit to having multiple types? As you already show, we import `Dummy1` via multiple paths, so why not import a single struct via multiple paths?